### ISSUE TITLE
Users provide ratings for users they have interacted with

### ISSUE THREAD

**Issue by shawnlauzon** [shawnlauzon/joatu#22]

**Comment by joatu** [comment]
In the short term, leaving ratings for everyone is a-ok.
In the longer term, having a drop-down of the user's offerings/requests/projects to select what the ratings is related to. Non-CAPS trades would make it otherwise difficult to verify a transaction happened or not.
Having the ratings you leave as visible (and tabulated in the average) can be a level you reach. Therefore, a new user couldn't spam someone's profile (good or bad), and spam accounts couldn't fill pages with advertisements/malware links.

**Comment by shawnlauzon** [comment]
What is the scenario you want to support for allowing a rating even if you have never had any interaction on the platform with the user?

I'm ok that we are not _verifying_ a transaction, as long as it is associated with some offer or request. Even if they didn't complete the transaction (with or without CAPS), posting a review I think is relevant.

**Comment by joatu** [comment]
How about this for minimum threshold for leaving a review : You have to have at least sent and received one message to and from the person on the site to be able to leave a rating.

**Comment by shawnlauzon** [comment]
I think that sounds reasonable.

Actually I think most of the conversations will be within the context of a POR. I actually think it would be nice to restrict messages to that (better tracking and analytics). Do you have a scenario that you want to support which involves sending a message independent of a POR?

**Comment by joatu** [comment]
How are you imagining structuring the P2P chat? If I see a user with many talents, I might go to their profile page and want to send them a general message to open up a conversation about creating a project together, or about some form of collaboration...or? So, yes, I suppose I do see conversations that can sit outside of the specific offers.

**Comment by shawnlauzon** [comment]
Yes, I agree unrestricted conversation does sound reasonable, and we should welcome that.

However, I continue to be concerned with ratings which are independent of a specific POR, even after a conversation has taken place. If these were just comments I wouldn't be worried. However, a 1..5 rating will be summarized in some way, which must have some level of trust associated with it (I remember this is one of the purposes of JoatU). But such a metric needs to be resilient to gaming, and right now I can't think of a good way to do that.

1. It seems like the number of reviews that can be left by any one person must be limited, but not zero. Because if I provide an offer, I should receive 0 or 1 review, no more. I have no idea how that can be enforced.
2. I can create multiple users and bump up their rating

If we provide a bad solution, users will have less trust in the rating in the future. Putting in a flawed or incomplete rating system is worse than no rating system at all.

I propose for MVP having no rating system, but allow users to leave as many comments as they want. In fact, without a rating I don't even think we need the requirement to have at least 1 message interchanged. We can investigate and improve as we move forward.

**Comment by joatu** [comment]
Allowing people to leave comments on the profiles of others (without the quantitative ratings) sounds like a good solution for the MVP.  Good thinking.

I do like the idea that people can select from a drop-down of what POR (including the option of general) they are leaving their comment about, but it isn't critical for the MVP.

We can work on elaborating the system of reviews/references as time goes on. I remember having a chat about the rating being broken up into several categories of 1-5 stars, promptness, respectfulness, etc.

**Comment by shawnlauzon** [comment]
Awesome. Teamwork!

**Comment by shawnlauzon** [comment]
Created #45 to handle the comments, returning this to the Inbox.
